debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/OracleDefaultValueConverter.java

```diff
@@ -0,0 +1,243 @@
+    public OracleDefaultValueConverter(OracleValueConverters valueConverters, OracleConnection jdbcConnection) {
+        this.valueConverters = valueConverters;
+        this.jdbcConnection = jdbcConnection;
+        this.defaultValueMappers = Collections.unmodifiableMap(createDefaultValueMappers());
+    }
+
+    @Override
+    public Optional<Object> parseDefaultValue(Column column, String defaultValue) {
+        final int dataType = column.jdbcType();
+        final DefaultValueMapper mapper = defaultValueMappers.get(dataType);
+        if (mapper == null) {
+            LOGGER.warn("Mapper for type '{}' not found.", dataType);
+            return Optional.empty();
+        }
+
+        try {
+            Object rawDefaultValue = mapper.parse(column, defaultValue);
+            Object convertedDefaultValue = convertDefaultValue(rawDefaultValue, column);
+            if (convertedDefaultValue instanceof Struct) {
+                // Workaround for KAFKA-12694
+                LOGGER.warn("Struct can't be used as default value for column '{}', will use null instead.", column.name());
+                return Optional.empty();
+            }
+            return Optional.ofNullable(convertedDefaultValue);
+        }
+        catch (Exception e) {
+            LOGGER.warn("Cannot parse column default value '{}' to type '{}'.  Expression evaluation is not supported.", defaultValue, dataType);
+            LOGGER.debug("Parsing failed due to error", e);
+            return Optional.empty();
+        }
+    }
+
+    private Object convertDefaultValue(Object defaultValue, Column column) {
+        // if converters is not null and the default value is not null, we need to convert default value
+        if (valueConverters != null && defaultValue != null) {
+            final SchemaBuilder schemaBuilder = valueConverters.schemaBuilder(column);
+            if (schemaBuilder != null) {
+                final Schema schema = schemaBuilder.build();
+                // In order to get the valueConverter for this column, we have to create a field;
+                // The index value -1 in the field will never used when converting default value;
+                // So we can set any number here;
+                final Field field = new Field(column.name(), -1, schema);
+                final ValueConverter valueConverter = valueConverters.converter(column, field);
+
+                Object result = valueConverter.convert(defaultValue);
+                if ((result instanceof BigDecimal) && column.scale().isPresent() && column.scale().get() > ((BigDecimal) result).scale()) {
+                    // Note as the scale is increased only, the rounding is more cosmetic.
+                    result = ((BigDecimal) result).setScale(column.scale().get(), RoundingMode.HALF_EVEN);
+                }
+                return result;
+            }
+        }
+        return defaultValue;
+    }
+
+    private Map<Integer, DefaultValueMapper> createDefaultValueMappers() {
```

Comment:
Can it be `static` (I prefer it that way, so to avoid any unintended access to potentially uninitialized instance fields from within this method when called by the constructor).